src/cli.rs

```diff
@@ -358,15 +358,12 @@ fn make_pkg_predicate(args: PackageSelectOptions) -> Result<Box<dyn Fn(&Package)
     }
 
     let publish = move |p: &Package| {
-        let publ = if ignore_publish {
-            true
-        } else if let Some(v) = p.publish() {
-            !v.is_empty()
-        } else {
-            true
-        };
-        trace!("{:}.publish={}", p.name(), publ);
-        publ
+        // If publish is set to false or any registry, it is ignored by default
+        // unless overriden.
+        let value = ignore_publish || p.publish().is_none();
```

Comment:
the `publish = ` flag is _unexepctedly complicated_ and this check incorrect. This would mark a package that has `publish = true` in the `Cargo.toml` as non-publish. That is because `= true` is a valid value according to the manifest and so is a vector of  registries. However, in the internal representation of `cargo`, this is captures to a `Option<Vec<String>>`, which should you leave wonedring: how is `publish = true`, represented?

Well, duh, `Some<vec![]>` of course. Obvi. And that was what the previous `!v.empty()` was checking for in case there _was_ a `Some(v)`. Removing that and checking only for _some_ value marks `publish = true` as falsely as non-publish.

This is confusing AF.

Reply:
I just tested that with
```rust
eprintln!("{:}.publish={:?}", p.name(), p.publish())
```

and got:
|                               |                                               |
|-------------------------------|-----------------------------------------------|
| no `publish`                  | cargo-unleash.publish=None                    |
| `publish = true`              | cargo-unleash.publish=None                    |
| `publish = false`             | cargo-unleash.publish=Some([])                |
| `publish = []`                | cargo-unleash.publish=Some([])                |
| `publish = ["some-registry"]` | cargo-unleash.publish=Some(["some-registry"]) |

So this seems to be in line with documentation for `Package::publish`, which says
```
/// Returns `None` if the package is set to publish.
/// Returns `Some(allowed_registries)` if publishing is limited to specified
/// registries or if package is set to not publish.
```
and so it looks like `p.publish().is_none()` should correctly tell us if a package is marked to publish in general (not restricted to a specific registry)

Reply:
@gnunicorn maybe this changed between 0.43.1 and 0.48 and is different now